tests/e2e/lib.sh

```diff
@@ -43,42 +43,71 @@ deploy_stackrox() {
+    sensor_regex="${helm_repo_name}/stackrox-secured-cluster-services[ \t]*.${SENSOR_CHART_VERSION_OVERRIDE}[ \t]*.([0-9]+\.[0-9]+\.[0-9]+)"
 
-    # Bounce collectors to avoid restarts on initial module pull
-    kubectl -n stackrox delete pod -l app=collector --grace-period=0
+    # If the central version is the latest the default behavior of deploy_central() is correct for compatibility tests
+    if  [[ $helm_charts =~ $central_regex ]]; then
+        ci_export CENTRAL_CHART_DIR_OVERRIDE "./central-chart-dir-compatibility-tests"
+        central_chart="${helm_repo_name}/stackrox-central-services"
+        helm pull "${central_chart}" --version "${CENTRAL_CHART_VERSION_OVERRIDE}" --untar --untardir "${CENTRAL_CHART_DIR_OVERRIDE}"
+    elif [[ "$latest_tag" != "$CENTRAL_CHART_VERSION_OVERRIDE" ]]; then
+        echo >&2 "stackrox-central-services helm chart for version ${CENTRAL_CHART_VERSION_OVERRIDE} not found in ${helm_repo_name} repo nor is it the latest tag."
+        exit 1
+    fi
 
-    sensor_wait
+    # If the sensor version is the latest the default behavior of deploy_sensor() is incorrect, because it will deploy
+    # a sensor version to match the central version. In our tests we want to test latest sensor vs older central too,
+    # and since latest sensor is not available in the repo either the chart is created here in the elif case.
+    if [[ $helm_charts =~ $sensor_regex ]]; then
+        sensor_chart="${helm_repo_name}/stackrox-secured-cluster-services"
+        ci_export SENSOR_CHART_DIR_OVERRIDE "./sensor-chart-dir-compatibility-tests"
+        helm pull "${sensor_chart}" --version "${SENSOR_CHART_VERSION_OVERRIDE}" --untar --untardir "${SENSOR_CHART_DIR_OVERRIDE}"
+    elif [[ "$latest_tag" == "$SENSOR_CHART_VERSION_OVERRIDE" ]]; then
+        if [[ $(roxctl version) != "$latest_tag" ]]; then
+            echo >&2 "Roxctl version $(roxctl version) is not equal to latest tag ${latest_tag}"
+            exit 1
+        fi
+        ci_export SENSOR_CHART_DIR_OVERRIDE "./sensor-chart-dir-compatibility-tests"
+        roxctl helm output secured-cluster-services --image-defaults=opensource --output-dir "${SENSOR_CHART_DIR_OVERRIDE}" --remove
+        echo "Downloaded stackrox-secured-cluster-services helm chart for version ${SENSOR_CHART_VERSION_OVERRIDE} to ${SENSOR_CHART_DIR_OVERRIDE}"
+        unset SENSOR_CHART_VERSION_OVERRIDE
+    else
+        echo >&2 "stackrox-secured-cluster-services helm chart for version ${SENSOR_CHART_VERSION_OVERRIDE} not found in ${helm_repo_name} repo nor is it the latest tag."
+        exit 1
+    fi
 
-    wait_for_collectors_to_be_operational
+    deploy_stackrox
 
-    touch "${STATE_DEPLOYED}"
+    if [[ -n "${CENTRAL_CHART_DIR_OVERRIDE}" ]]; then
+        rm -r "${CENTRAL_CHART_DIR_OVERRIDE}"
+    fi
+    if [[ -n "${SENSOR_CHART_DIR_OVERRIDE}" ]]; then
+        rm -r "${SENSOR_CHART_DIR_OVERRIDE}"
+    fi
```

Comment:
Don't hesitate `rm -rf`. This version doesn't prompt to delete readonly and other special files.

```suggestion
    if [[ -n "${CENTRAL_CHART_DIR_OVERRIDE}" ]]; then
        rm -rf "${CENTRAL_CHART_DIR_OVERRIDE}"
    fi
    if [[ -n "${SENSOR_CHART_DIR_OVERRIDE}" ]]; then
        rm -rf "${SENSOR_CHART_DIR_OVERRIDE}"
    fi
```